File: frontend/src/types/email.ts
export interface Email {
  id: string;
  thread_id: string;
  message_id: string;
  sender: string;
  sender_name: string | null;
  subject: string;
  body: string;
  direction: 'inbound' | 'outbound';
  heuristic_result: HeuristicResult | null;
  classification: Classification | null;
  confidence_score: number | null;
  priority_score: number | null;
  is_processed: boolean;
  requires_human: boolean;
  received_at: string;
  created_at: string;
}

export interface HeuristicResult {
  is_spam: boolean;
  is_ransomware: boolean;
  is_legal_threat: boolean;
  is_gdpr_request: boolean;
  is_internal: boolean;
  is_urgent: boolean;
  do_not_auto_reply: boolean;
  urgency_score: number;
  spam_score: number;
  flags: string[];
}

export interface Classification {
  category: string;
  sentiment: string;
  sentiment_score: number;
  urgency: string;
  confidence: number;
  requires_human: boolean;
  escalation_reason: string | null;
  suggested_reply: string | null;
  entities: Record<string, unknown>;
  summary: string;
  risk_indicators: string[];
}

export interface EmailListResponse {
  emails: Email[];
  total: number;
  page: number;
  page_size: number;
  has_next: boolean;
}

File: backend/app/schemas/email.py
"""
SentinelAI — Email Schemas

Pydantic models for email ingestion, responses, and classification.
"""

from datetime import datetime
from pydantic import BaseModel, Field, EmailStr
from uuid import UUID


class EmailIngestPayload(BaseModel):
    """Schema for a single email in the ingestion endpoint."""
    message_id: str = Field(..., description="Unique email Message-ID header")
    in_reply_to: str | None = Field(None, description="Message-ID of parent email")
    references: str | None = Field(None, description="Space-separated Message-IDs of thread")
    sender: str = Field(..., description="Sender email address")
    sender_name: str | None = Field(None, description="Sender display name")
    recipients: list[str] = Field(default_factory=list, description="Recipient addresses")
    subject: str = Field(..., min_length=1, description="Email subject")
    body: str = Field(..., min_length=1, description="Email body text")
    body_html: str | None = Field(None, description="HTML version of body")
    headers: dict | None = Field(default_factory=dict, description="Raw email headers")
    received_at: datetime | None = Field(None, description="When the email was received")


class EmailBatchIngest(BaseModel):
    """Batch email ingestion request."""
    emails: list[EmailIngestPayload] = Field(..., min_length=1, max_length=100)


class EmailResponse(BaseModel):
    """Standard email response."""
    id: UUID
    thread_id: UUID
    message_id: str
    sender: str
    sender_name: str | None
    subject: str
    body: str
    direction: str
    heuristic_result: dict | None
    classification: dict | None
    confidence_score: float | None
    priority_score: float | None
    is_processed: bool
    requires_human: bool
    received_at: datetime
    created_at: datetime

    class Config:
        from_attributes = True


class EmailListResponse(BaseModel):
    """Paginated email list."""
    emails: list[EmailResponse]
    total: int
    page: int
    page_size: int
    has_next: bool
